src/Optional.php

```diff
@@ -171,21 +171,25 @@ public function orElse($other)
      *
      * @throws $exception 存在する値がない場合
      * @throws \RuntimeException $exception 指定した例外クラスが存在しない場合
+     *
+     * @see isPresent()
+     * @see get()
      */
     public function orElseThrow(string $exception)
     {
+        if ($this->isPresent()) {
+            return $this->get();
+        }
+
         if (!class_exists($exception)) {
             throw new \RuntimeException("'{$exception}' exception does not exist.");
         }
-        if (is_null($this->value)) {
-            $e = new $exception;
-            if (!($e instanceof \Throwable)) {
-                throw new \RuntimeException("'{$exception}' not implement Throwable.");
-            }
-            throw $e;
-        }
 
-        return $this->value;
+        $e = new $exception;
```

Comment:
__[phpmd]__
* Avoid variables with short names like $e. Configured minimum length is 3.